### PR TITLE
fix:trace graceful stop,pre loss trace

### PR DESCRIPTION
**Pull request by wsx864321**

### CODE REVIEW

File: core/trace/agent.go [hunk 1]
```diff
@@ -43,6 +45,11 @@ func StartAgent(c Config) {
 	agents[c.Endpoint] = lang.Placeholder
 }
 
+// StopAgent shuts down the span processors in the order they were registered.
+func StopAgent() {
+	_ = tp.Shutdown(context.TODO())
```
Comment (kevwan) [inline core/trace/agent.go:50]: Use Background()

File: core/trace/agent.go [hunk 2]
```diff
@@ -74,7 +81,7 @@ func startAgent(c Config) error {
 		opts = append(opts, sdktrace.WithBatcher(exp))
 	}
 
-	tp := sdktrace.NewTracerProvider(opts...)
+	tp = sdktrace.NewTracerProvider(opts...)
```
Comment (kevwan) [inline core/trace/agent.go:84]: AddShutdownListener in this func?

**Review by kevwan — CHANGES_REQUESTED** [review]

File: core/trace/agent.go (hunk 1 above)
Comment (wsx864321) [reply core/trace/agent.go:50]: Uh-huh,Background() is better as the top-level Context.

File: core/trace/agent.go (hunk 2 above)
Comment (wsx864321) [reply core/trace/agent.go:84]: I'll revise it

**Review by kevwan — APPROVED** [review]
lgtm

**Review by re-dylan — APPROVED** [review]
LGTM

**Review by re-dylan — APPROVED** [review]

**Pull request merged**